tests/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,279 @@
+# Consumers of this makefile are expected to set the relevant variables
+# e.g. ENTRY, CBMCFLAGS, UNWINDSET etc
+# and then include this makefile.  To see how to use this makefile, take a look
+# at any of the existing proofs in this repository.
+#
+# This makefile is designed so that all files created during a proof are
+# placed in the proof directory, or subdirectories of it.
+# In particular, all gotos are built into the "gotos" folder, and logs go into
+# the "logs" folder. This means that the main source directories are untouched by
+# doing a build, and should make it safe to simultaniously build multiple proofs,
+# including proofs with different configurations and -D defines.
+#
+# Dependency handling in this file may not be perfect.  It is recommended that
+# you "make clean" or "make veryclean" before rerunning a proof.
+
+SHELL=/bin/bash
+
+default: report
+
+# if Makefile.local exists, use it. This provides a way to override the defaults
+sinclude Makefile.local
+
+################################################################
+# Define some constants that are hard to reference otherwise
+SPACE :=$() $()
+COMMA :=,
+
+################################################################
+# Paths to binaries
+# By default, use the version on your path; this can be overriden to select particular
+# versions of the tools. /usr/bin/goto-analyzer or ${HOME}/sw/goto-analyzer etc
+GOTO_ANALYZER ?= goto-analyzer
+GOTO_CC ?= goto-cc
+GOTO_INSTRUMENT ?= goto-instrument
+VIEWER ?= cbmc-viewer
+
+################################################################
+# Setup directory aliases
+SRCDIR ?= $(abspath ../../../..)
+HELPERDIR ?= $(SRCDIR)/tests/cbmc
+PROOFDIR = $(abspath .)
+GOTODIR = $(PROOFDIR)/gotos
+LOGDIR = $(PROOFDIR)/logs
+
+################################################################
+# Useful macros for running commands
+#1: command, 2: flags, 3: log file
+DO_AND_LOG_COMMAND = $(1) $(2) 2>&1 | tee $(3); exit $${PIPESTATUS[0]}
+
+#1: flags, 2: source, 3: target
+DO_GOTO_ANALYZER = $(call DO_AND_LOG_COMMAND,$(GOTO_ANALYZER),$(1) $(2) $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: flags, 2: source, 3: target
+DO_GOTO_CC =  $(call DO_AND_LOG_COMMAND,$(GOTO_CC),$(1) $(2) -o $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: flags, 2: source, 3: target
+DO_GOTO_INSTRUMENT = $(call DO_AND_LOG_COMMAND,$(GOTO_INSTRUMENT),$(1) $(2) $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: message 2: source 3: dest 
+DO_NOOP_COPY = cp $(2) $(3); echo $(1) | tee $(call LOG_FROM_ENTRY,$(3))
+
+################################################################
+# Useful macros translating filenames
+C_FROM_GOTO = $(patsubst $(GOTODIR)%,$(SRCDIR)%,$(1:.goto=.c))
+GOTO_FROM_C = $(patsubst $(SRCDIR)%,$(GOTODIR)%,$(1:.c=.goto))
+LOG_FROM_ENTRY = $(LOGDIR)/$(notdir $(1:.goto=.log))
+LOG_FROM_GOTO = $(patsubst $(GOTODIR)%,$(LOGDIR)%,$(1:.goto=.log))
+
+################################################################
+# Default CBMC flags
+CBMC_VERBOSITY ?= ""
```

Comment:
Suggest `CBMC_VERBOSITY ?= "--verbosity 0"` or 1 to document how people should override it

Reply:
Done